### PR TITLE
Feature notifications

### PR DESCRIPTION
**Pull request by The-Vector-007**

### CODE REVIEW

**Comment by Ashwin-op** [comment]
Can you please change the base branch to [this](https://github.com/commudle/commudle-ng/tree/feature-notifications)?

**Comment by The-Vector-007** [comment]
> Can you please change the base branch to [this](https://github.com/commudle/commudle-ng/tree/feature-notifications)?

done!

File: projects/shared-models/notifications.model.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { INotification } from './notification.model';
+
+export interface INotifications {
+  notifications: INotification[];
+  count: Number;
```
Comment (Ashwin-op) [inline projects/shared-models/notifications.model.ts:5]: shouldn't these be `number` type?

Comment (Ashwin-op) [inline projects/shared-models/notifications.model.ts]: shouldn't these be `number` type?

File: projects/shared-models/notification.model.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import { ILab } from './lab.model';
+import { IUser } from './user.model';
+import { IUserMessage } from './user_message.model';
+
+export interface INotification {
+  entity: IUser | ILab | ICommunityBuild | IUserMessage;
+  sender: IUser;
+  id: string;
+  notification_message: string;
+  status: string;
```
Comment (Ashwin-op) [inline projects/shared-models/notification.model.ts]: This should be the corresponding `enum` type right?

File: projects/commudle-admin/src/app/app-routing.module.ts [hunk 3]
```diff
@@ -28,6 +29,10 @@ const routes = [
     path: 'about',
     component: AboutOldComponent,
   },
+  {
+    path: 'notifications',
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/app-routing.module.ts:33]: Let's convert this into a module, so that it's lazy loaded

File: projects/commudle-admin/src/app/feature-modules/notifications/services/notification.service.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+})
+export class NotificationService {
+  constructor(private apiRoutesService: ApiRoutesService, private http: HttpClient) {}
+
+  getAllNotifications(page, count): Observable<INotifications> {
+    const params = new HttpParams().set('page', page).set('count', count);
+    return this.http.get<INotifications>(this.apiRoutesService.getRoute(API_ROUTES.NOTIFICATIONS.INDEX), { params });
+  }
+
+  updateNotificationStatus(status: string, id: string) {
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/notifications/services/notification.service.ts]: status is of the corresponding `enum` type

File: projects/commudle-admin/src/app/feature-modules/notifications/services/notification.service.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    const params = new HttpParams().set('notification_queue_id', id);
+    return this.http.post<any>(
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/notifications/services/notification.service.ts]: Let's not use `any`

File: projects/commudle-admin/src/app/feature-modules/notifications/services/websockets/notification-channel.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  actionCable = actionCable;
+  private cableConnection;
+  private subscription;
+
+  private notificationData: BehaviorSubject<any> = new BehaviorSubject(null);
+  public notificationData$: Observable<any> = this.notificationData.asObservable();
+
+  constructor(
+    private actionCableConnection: ActionCableConnectionSocket,
+    private authWatchService: LibAuthwatchService,
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/notifications/services/websockets/notification-channel.ts]: is this needed here?

File: projects/commudle-admin/src/app/feature-modules/notifications/services/websockets/notification-channel.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        channel: APPLICATION_CABLE_CHANNELS.NOTIFICATION_CHANNEL,
+      },
+      {
+        received: (data) => {
+          this.notificationData.next(data);
+        },
+      },
+    );
+
+    return this.subscription;
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/notifications/services/websockets/notification-channel.ts]: is return needed here?

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+  ngOnDestroy(): void {
+    this.subscriptions.forEach((subscription) => {
+      subscription.unsubscribe();
+    });
+  }
+
+  getOlderNotifications() {
+    this.notificationService.getAllNotifications(this.page, this.count).subscribe((val) => {
+      this.notifications = val.notifications;
+      console.log(this.notifications);
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts]: let's remove console.log

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+  markAsRead(notification: INotification) {
+    this.changeStatus(ENotificationStatuses.READ, notification);
+  }
+
+  markAsInteracted(notification: INotification) {
+    this.changeStatus(ENotificationStatuses.INTERACTED, notification);
+  }
+
+  changeStatus(status: string, notification: INotification) {
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts]: status is of the enum type

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+  receiveData() {
+    this.subscriptions.push(
+      this.notificationChannel.notificationData$.subscribe((data) => {
+        if (data) {
+          switch (data.action) {
+            case this.notificationChannel.ACTIONS.NEW_NOTIFICATION: {
+              this.notifications.unshift(data?.notification);
+            }
+            case this.notificationChannel.ACTIONS.STATUS_UPDATE: {
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts:50]: why is this case blank?

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+  markAsInteracted(notification: INotification) {
+    this.changeStatus(ENotificationStatuses.INTERACTED, notification);
+  }
+
+  changeStatus(status: string, notification: INotification) {
+    this.notificationService.updateNotificationStatus(status, notification.id).subscribe((val) => {
+      if (val) {
+        let idx = -1;
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts]: Let's make this block of code a little better

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.html [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+            alt="{{ notification.sender.name }}"
+            src="{{ notification.sender.avatar }}"
+            title="{{ notification.sender.name }}"
+          />
+        </div>
+        <div class="content">
+          {{ notification.notification_message }}
+        </div>
+        <nb-icon
+          *ngIf="notification.status === 'unread'"
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.html]: let's use enum here

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.html [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+          position="left center"
+        ></nb-badge>
+        <div class="avatar">
+          <img
+            alt="{{ notification.sender.name }}"
+            src="{{ notification.sender.avatar }}"
+            title="{{ notification.sender.name }}"
+          />
+        </div>
+        <div class="content">
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.html]: let's use `p` tags for content

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.scss [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+
+          nb-icon {
+            visibility: visible;
+          }
+        }
+
+        .avatar {
+          border-radius: 50%;
+          max-height: 56px;
+          max-width: 56px;
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.scss]: let's use tailwindcss here

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.html [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+          *ngIf="notification.status === 'unread'"
+          (click)="markAsRead(notification)"
+          [icon]="'checkmark-outline'"
+          [status]="'basic'"
+        ></nb-icon>
+      </nb-list-item>
+    </nb-list>
+  </nb-card-body>
+  <nb-card-footer>
+    <!-- TODO: change it to commudle.com -->
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.html]: let's change this to relative url

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.scss [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+  border: 0;
+
+  nb-card-body {
+    padding: 0;
+    nb-list {
+      nb-list-item {
+        position: relative;
+
+        &:hover {
+          background-color: honeydew;
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.scss]: let's not use new colours, either we use colours we have already defined or from tailwind

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+import { NotificationService } from 'projects/commudle-admin/src/app/feature-modules/notifications/services/notification.service';
+import { ENotificationStatuses } from 'projects/shared-models/enums/notification_statuses.enum';
+
+@Component({
+  selector: 'app-notifications-popover',
+  templateUrl: './notifications-popover.component.html',
+  styleUrls: ['./notifications-popover.component.scss'],
+})
+export class NotificationsPopoverComponent implements OnInit, OnDestroy {
+  subscriptions = [];
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts:13]: are we using this array?

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.ts [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+import { NotificationChannel } from 'projects/commudle-admin/src/app/feature-modules/notifications/services/websockets/notification-channel';
+import { INotification } from 'projects/shared-models/notification.model';
+
+@Component({
+  selector: 'app-notifications-page',
+  templateUrl: './notifications-page.component.html',
+  styleUrls: ['./notifications-page.component.scss'],
+})
+export class NotificationsPageComponent implements OnInit, OnDestroy {
+  subscriptions = [];
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.ts:12]: are we using this array?

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.ts [hunk 19]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  receiveData() {
+    this.subscriptions.push(
+      this.notificationChannel.notificationData$.subscribe((data) => {
+        switch (data?.action) {
+          case this.notificationChannel.ACTIONS.NEW_NOTIFICATION: {
+            this.notifications.unshift(data?.notification);
+          }
+          case this.notificationChannel.ACTIONS.STATUS_UPDATE: {
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.ts]: are we using this case?

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.ts [hunk 20]
```diff
@@ -0,0 +1,62 @@
...
+      });
+    }
+  }
+
+  receiveData() {
+    this.subscriptions.push(
+      this.notificationChannel.notificationData$.subscribe((data) => {
+        switch (data?.action) {
+          case this.notificationChannel.ACTIONS.NEW_NOTIFICATION: {
+            this.notifications.unshift(data?.notification);
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.ts]: why are we checking here if `data` is defined?

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts [hunk 21]
```diff
@@ -0,0 +1,80 @@
...
+    });
+  }
+
+  receiveData() {
+    this.subscriptions.push(
+      this.notificationChannel.notificationData$.subscribe((data) => {
+        if (data) {
+          switch (data.action) {
+            case this.notificationChannel.ACTIONS.NEW_NOTIFICATION: {
+              this.notifications.unshift(data?.notification);
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts]: why are we checking here if `data` is defined?

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.scss [hunk 22]
```diff
@@ -0,0 +1,80 @@
...
+  max-width: 768px;
+
+  nb-card {
+    nb-card-body {
+      padding: 1rem 1rem;
+
+      nb-list {
+        nb-list-item {
+          .avatar {
+            border-radius: 50%;
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.scss]: Let's use tailwind classes for this

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.scss [hunk 23]
```diff
@@ -0,0 +1,80 @@
+@import './projects/commudle-admin/src/assets/styles/variables';
+
+.container {
+  max-width: 768px;
+
+  nb-card {
+    nb-card-body {
+      padding: 1rem 1rem;
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.scss]: this can be shortened

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.scss [hunk 24]
```diff
@@ -0,0 +1,80 @@
+@import './projects/commudle-admin/src/assets/styles/variables';
+
+.container {
+  max-width: 768px;
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.scss]: any reason?

File: projects/commudle-admin/src/app/components/navbar-menu/navbar-menu.component.ts [hunk 25]
```diff
@@ -1,22 +1,64 @@
...
 
-  ngOnInit(): void {}
+  notificationsPopoverComponent = NotificationsPopoverComponent;
+
+  subscriptions = [];
+
+  constructor(private notificationChannel: NotificationChannel) {}
+
+  ngOnInit(): void {
+    if (/Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent)) {
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/components/navbar-menu/navbar-menu.component.ts:32]: Why is this needed?

File: projects/commudle-admin/src/app/components/navbar-menu/navbar-menu.component.ts [hunk 26]
```diff
@@ -1,22 +1,64 @@
...
+  faBell = faBell;
 
-  constructor() {}
+  mobileView: boolean;
+  notificationCount = 0;
 
-  ngOnInit(): void {}
+  notificationsPopoverComponent = NotificationsPopoverComponent;
+
+  subscriptions = [];
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/components/navbar-menu/navbar-menu.component.ts:27]: are we using this array?

File: projects/commudle-admin/src/app/components/navbar-menu/navbar-menu.component.html [hunk 27]
```diff
@@ -5,12 +5,36 @@
...
+
+  <a *ngIf="mobileView" [routerLink]="['/notifications']" class="notification">
+    <div class="icon">
+      <span *ngIf="notificationCount > 0" class="notification-badge">{{ notificationCount }}</span>
+      <fa-icon [icon]="faBell"></fa-icon>
+    </div>
+    <div class="home-navbar-menu-option">Notifications</div>
+  </a>
+
+  <a *ngIf="!mobileView" class="notification">
```
Comment (Ashwin-op) [inline projects/commudle-admin/src/app/components/navbar-menu/navbar-menu.component.html:24]: Let's think of a better way to do this

**Comment by Ashwin-op** [comment]
- on clicking a notification, let's redirect to that specific page
- let's remove the red colour left border and also the light green background, instead a normal light grey background works (with a fade-in transition). Also a little bit of padding-left would look better
image
- The notification icon should be highlighted on this page
image
- breaking in mobile view
image
- count not resetting in mobile view
image

**Review by Ashwin-op — CHANGES_REQUESTED** [review]

File: projects/shared-models/notifications.model.ts (hunk 1 above)
Comment (The-Vector-007) [reply projects/shared-models/notifications.model.ts]: autocomplete issue. Converted to number.

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts (hunk 10 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts:50]: shift change status code to this switch case

File: projects/commudle-admin/src/app/feature-modules/notifications/services/notification.service.ts (hunk 5 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/notifications/services/notification.service.ts]: change it to boolean

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.scss (hunk 24 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.scss]: remove max-width, add to clickup for one page layout

File: projects/shared-models/notification.model.ts (hunk 2 above)
Comment (The-Vector-007) [reply projects/shared-models/notification.model.ts]: done

File: projects/commudle-admin/src/app/app-routing.module.ts (hunk 3 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/app-routing.module.ts:33]: converted.

File: projects/commudle-admin/src/app/feature-modules/notifications/services/notification.service.ts (hunk 4 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/notifications/services/notification.service.ts]: done

File: projects/commudle-admin/src/app/feature-modules/notifications/services/notification.service.ts (hunk 5 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/notifications/services/notification.service.ts]: done

File: projects/commudle-admin/src/app/feature-modules/notifications/services/websockets/notification-channel.ts (hunk 6 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/notifications/services/websockets/notification-channel.ts]: nope, removed!

File: projects/commudle-admin/src/app/feature-modules/notifications/services/websockets/notification-channel.ts (hunk 7 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/notifications/services/websockets/notification-channel.ts]: nope, removed!

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts (hunk 8 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts]: removed!

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts (hunk 9 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts]: changed

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.html (hunk 12 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.html]: done!

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.html (hunk 13 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.html]: done!

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.html (hunk 15 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.html]: done!

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.scss (hunk 16 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.scss]: sure.

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts (hunk 17 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts:13]: yes

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.ts (hunk 18 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.ts:12]: yes

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.ts (hunk 19 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.ts]: yes, we will use that.

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.ts (hunk 20 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.ts]: used if instead

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts (hunk 21 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts]: used if instead

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.scss (hunk 23 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.scss]: done!

File: projects/commudle-admin/src/app/components/navbar-menu/navbar-menu.component.ts (hunk 26 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/components/navbar-menu/navbar-menu.component.ts:27]: yes

File: projects/commudle-admin/src/app/components/navbar-menu/navbar-menu.component.html (hunk 27 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/components/navbar-menu/navbar-menu.component.html:24]: Going with this for now, searching for other options.

File: projects/commudle-admin/src/app/components/navbar-menu/navbar-menu.component.ts (hunk 25 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/components/navbar-menu/navbar-menu.component.ts:32]: To differentiate between desktop and mobile view.

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.scss (hunk 14 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.scss]: done!

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.scss (hunk 22 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-page/notifications-page.component.scss]: done!

File: projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts (hunk 11 above)
Comment (The-Vector-007) [reply projects/commudle-admin/src/app/feature-modules/notifications/components/notifications-popover/notifications-popover.component.ts]: done!

**Comment by The-Vector-007** [comment]
TODO: Detection of mobile view to be checked in SSR.

**Comment by Ashwin-op** [comment]
merged changes to test-server branch
